### PR TITLE
CONSOLE-2447: i18n for secrets

### PR DESCRIPTION
**Pull request by zherman0**
[CONSOLE-2447](https://issues.redhat.com/browse/CONSOLE-2447): i18n for secrets
Needs #7045 to merge to be complete.

Also includes a missed translation from the details page component that was found during my testing.

### CODE REVIEW

File: frontend/public/components/secret.jsx [hunk 1]
```diff
@@ -34,17 +36,20 @@ const actionButtons = [addSecretToWorkload];
...
-    accessReview: {
-      group: kind.apiGroup,
-      resource: kind.plural,
-      name: obj.metadata.name,
-      namespace: obj.metadata.namespace,
-      verb: 'update',
-    },
-  }),
+  (kind, obj) => {
+    const { t } = useTranslation();
```
Comment (spadgett) [inline frontend/public/components/secret.jsx:40]: You shouldn't be calling `useTranslation` here. It's only to be used inside a component. Return a `labelKey` instead.

File: frontend/public/components/secret.jsx [hunk 2]
```diff
@@ -208,12 +214,13 @@ const filters = [
...
 
 const SecretsPage = (props) => {
+  const { t } = useTranslation();
   const createItems = {
-    generic: 'Key/Value Secret',
-    image: 'Image Pull Secret',
-    source: 'Source Secret',
-    webhook: 'Webhook Secret',
-    yaml: 'From YAML',
+    generic: t('workload~Key/Value secret'),
```
Comment (spadgett) [inline frontend/public/components/secret.jsx:219]:
```suggestion
    generic: t('workload~Key/value secret'),
```

File: frontend/public/components/secret.jsx [hunk 3]
```diff
@@ -208,12 +214,13 @@ const filters = [
...
+    image: t('workload~Image Pull secret'),
```
Comment (spadgett) [inline frontend/public/components/secret.jsx:220]:
```suggestion
    image: t('workload~Image pull secret'),
```

File: frontend/public/components/secret.jsx [hunk 4]
```diff
@@ -25,7 +26,8 @@ export const addSecretToWorkload = (kindObj, secret) => {
 
   return {
     callback: () => configureAddSecretToWorkloadModal({ secretName, namespace, blocking: true }),
-    label: 'Add Secret to Workload',
+    // t('workload~Add secret to workload')
+    labelKey: 'workload~Add secret to workload',
```
Comment (spadgett) [inline frontend/public/components/secret.jsx:30]: I believe we're using the kind labels here, so Secret would be capitalized.

```suggestion
    labelKey: 'workload~Add Secret to workload',
```

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals
/assign @yapei @ahardin-rh @sferich888

**Comment by yanpzhan** [comment]
@zherman0   Tested against the pr, on secret details page, the "Add Secret to workload" button is now an empty small button and "Data" is not marked for translation, could you check if it need update?

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/lgtm cancel

@zherman0 It looks like the CRUD tests need to be updated for the secrets change.

**Comment by spadgett** [comment]
> edits the resource instance: Kubernetes resource CRUD operations Secret edits the resource instance
> AssertionError: Timed out retrying: Expected to find element: `[data-test-action="Edit Secret"]:not(.pf-m-disabled)`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-c23lsyfm-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/crud/k8-openshift-cruds.spec.ts:37306:57)

**Comment by zherman0** [comment]
@yanpzhan - Thanks for pointing those out; they should be resolved now.

**Comment by zherman0** [comment]
@spadgett - This PR is now passing all test and ready for review

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by zherman0** [comment]
@spadgett - had to rebase due to json conflict

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/hold

It looks like the secret tests aren't passing.

**Comment by zherman0** [comment]
@spadgett - Been working on this all morning; the tests all pass locally so I am not sure what if happening.

**Comment by zherman0** [comment]
@spadgett - Fixed the test cases and rebased.

File: frontend/packages/integration-tests-cypress/views/list-page.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ export const listPage = {
     cy.byLegacyTestID('namespace-bar-dropdown').should('not.exist'),
   clickCreateYAMLdropdownButton: () => {
     return cy
-      .byLegacyTestID('dropdown-button')
+      .get('#item-create')
```
Comment (dtaylor113) [inline frontend/packages/integration-tests-cypress/views/list-page.ts:11]: Hi, I know I said this was ok, but I just found where you can add a `data-test="item-create"` [here](https://github.com/openshift/console/blob/1c9b925b8d722e9781a60c11fe39258d90bf916a/frontend/public/components/factory/list-page.jsx#L220).
Then you can use `.byTestID('item-create')...`.  
Thanks

**Review by dtaylor113 — CHANGES_REQUESTED** [review]
Hi, one small Cypress fix -thanks

File: frontend/packages/integration-tests-cypress/views/list-page.ts (hunk 5 above)
Comment (dtaylor113) [reply frontend/packages/integration-tests-cypress/views/list-page.ts:11]: Can you pls also remove the `return` in `return cy..`?  -thanks

Comment (zherman0) [reply frontend/packages/integration-tests-cypress/views/list-page.ts]: Okay, updated the code.

**Review by dtaylor113 — APPROVED** [review]
/lgtm

**Comment by zherman0** [comment]
/retest

So far we have had annotation flakes and a variety of others.

**Pull request merged**